src/Cli/dotnet/commands/dotnet-test/IPC/Serializers/TestResultMessagesSerializer.cs

```diff
@@ -252,13 +252,44 @@ private static List<FailedTestResultMessage> ReadFailedTestMessagesPayload(Strea
                             reason = ReadStringValue(stream, fieldSize);
                             break;
 
-                        case FailedTestResultMessageFieldsId.ErrorMessage:
-                            errorMessage = ReadStringValue(stream, fieldSize);
-                            break;
+                        case FailedTestResultMessageFieldsId.ExceptionMessageList:
```

Comment:
It's better we extract this part into another method.

Reply:
https://github.com/dotnet/sdk/pull/45398 will address here